psi4/src/psi4/dct/dct.h

```diff
@@ -320,9 +338,9 @@ class DCTSolver : public Wavefunction {
     /// The maximum number of IDPs possible for cumulant updates
     int dim_cumulant_;
     /// The lookup array that determines which compound indices belong to orbital IDPs and which don't
-    int *lookup_orbitals_;
+    std::vector<bool> lookup_orbitals_;
```

Comment:
Be very careful with this data structure - under the hood each element is a single bit, which means it's nicely optimized for size.  However you can get crazy race conditions as a result - if 2 threads are trying to assign different elements within the same byte, trouble can happen.  If you're not writing to this array  in parallel anywhere all is well, but it's worth being aware of.

Reply:
Thanks for the warning! The code to parallelize assignments to this structure are specifically commented out in the existing code, so there are no parallelization bugs in the actually used code. The code in question is "suboptimal" and would need to be revisited to make it useful, anyways.

"suboptimal" = does second-order convergence for DCFT-06 only, which is prone to convergence problems